### DISCUSSION TITLE
Bookworm out in June.... new Spiral ISOs???

### DISCUSSION THREAD

**Discussion by methanoid** [SpiralLinux/SpiralLinux-project#256] [General]
https://www.phoronix.com/news/Debian-12.0-Release-Date

Will save all that time doing a manual upgrade on install to Bookworm :-)

**Comment by robertocannito** [comment]
Dear Sam, all us here are waiting for your answer ;)

**Comment by methanoid** [comment]
We probably should have tagged @geckolinux  ;-)

I'm rebuilding for now with CUBIC, to make it Bookworm from get go... but the proper answer is probably to build our own with Debian LiveBuild, or wait till June

**Comment by CookieLemur** [comment]
Looking forward to switching to Debian with Spiral Linux once the new ISO comes out :)

**Reply by CicadaSeventeen** [reply]
You can just use my fork now.

**Reply by CicadaSeventeen** [reply]
see
https://github.com/DebianEasy/SpiralLinux-fork-bookworm

**Reply by InvisibleRasta** [reply]
any guide around on how to build the iso from those tar files?

**Reply by methanoid** [reply]
Its LiveBuild... look for the man pages or go watch a few YT videos on LiveBuild. Thats what I did

**Reply by CicadaSeventeen** [reply]
Installing live-build, untar the tar file, running `mkiso` scirpt

**Comment by methanoid** [comment]
ISO downloads?

**Reply by CicadaSeventeen** [reply]
Not yet, you need to use live-build to build iso yourself.

**Reply by zparihar** [reply]
Can you point me to where the instructions on how to build the ISO are located?

**Reply by CicadaSeventeen** [reply]
> Can you point me to where the instructions on how to build the ISO are located?

Iso is available now. See https://debianeasy.github.io

**Reply by CicadaSeventeen** [reply]
Kde Gnome Xfce Budgie and Cinnamon provided. If you want other spin, you can tell me.

**Reply by methanoid** [reply]
> Kde Gnome Xfce Budgie and Cinnamon provided. If you want other spin, you can tell me.

So does your fork use different wallpapers etc (seems you use the Debian logo for example)?
Have you checked the Debian people are happy with you using their name (DebianEasy is directly including their TM)?
You might want to fix the various spelling mistakes on your hub to make it look more professional ;-)

I'll give the KDE version a look 👍

**Reply by newbie992** [reply]
@CicadaSeventeen 
I downloaded and installed your debianeasy spin of SpiralLinux ( Kde ).
I like it very much.
Thank you for sharing it.

**Comment by methanoid** [comment]
@geckolinux   I appreciate new ISOs will take time (and obviously keen now D 12 is OUT!) but you may have seen the Learn Linux TV youtuber channel did a Debian 12 video about 12 "important" tweaks to do post install. I think every one of them was redundant if you have Spiral Linux... so clearly your choices are sane sensible ones.

Can't wait for Spiral KDE 12 :-)  Thank you

**Reply by geckolinux** [reply]
Nice! Thanks for saying so.
One potential bottleneck will be the https://fasttrack.debian.net repo, which is needed for VirtualBox guest support. (I personally hate VirtualBox, but it's what everyone uses and expects to work out-of-the-box.) I'm not sure how long it will take them to get the Fast Track repos up for Bookworm, but it will eventually be available.

**Reply by methanoid** [reply]
LOL, I also hate VirtualBox. Only virtualisation I would use would be KVM but understand if we have to wait for VB support :-(

**Reply by CookieLemur** [reply]
I second this! Thanks for the reply and your hard work dev, know that it is very much appreciated :)

**Reply by fweller** [reply]
I use VirtualBox on all of my computers, and use it every day.  I stopped pulling VirtualBox from the repos and instead install directly from the website download.  Why?  For two reasons.  1, support is only provided to those using the official VB downloads.  2, the "All Distributions" install (ends with a .run filename extension) has installed on any distro I have ever used, so I don't need to look for a distro and version specific install.

**Reply by geckolinux** [reply]
@fweller In the case of SpiralLinux I'm not as concerned about being able to use it as a VirtualBox host, but rather as a guest. Almost everyone initially tests distros as VirtualBox guests, and out-of-the-box support for the VirtualBox guest module (for clipboard integration, automatic screen resizing, etc.) is expected. So that really needs to be available in the repositories to be integrated with SpiralLinux during the ISO build process.

**Reply by fweller** [reply]
@geckolinux what you say makes a lot of sense.  Many people, myself included, test a new distro out in a VM, and it is very convenient if the distro supports VirtualBox client by default.  Many distro's do support it, Ubuntu does not, and as I just tested Debian 12 Bookworm live non-free, it also does not.  I appreciate that you are maintaining a level of quality of your distro to keep VirtualBox client support available.  Perhaps a compromise to all of your Spiral fans is to release an ISO now, and add in virtualbox support when Debian makes the repo available.

**Reply by methanoid** [reply]
I agree...  the longer there is no Spiral release the more people find alternatives or spin their own or lose interest.  A version without VB support that clearly says DONT test in VB cos it wont work would do the job. 

(sent from my Debian 12 non Spiral install :-( cos there is still no Spiral 12)

**Reply by banzaigtv** [reply]
I'm already looking at Q4OS 5, MX Linux 23, and LMDE 6 as alternatives, but I will probably hang on to the current Debian 11 version.

**Reply by geckolinux** [reply]
@methanoid @banzaigtv I guess I don't see what the problem is, if you're already using and liking SpiralLinux then why not simply upgrade it to Debian 12?
https://github.com/SpiralLinux/SpiralLinux-project/wiki#upgrading-to-a-major-new-debian-stable-release
The resulting upgraded installation will be exactly the same as installing a new system from a future SpiralLinux ISO built from Debian 12. SpiralLinux is meant to be installed once and then infinitely updated/upgraded directly from Debian while maintaining the SpiralLinux tweaks.

**Reply by banzaigtv** [reply]
@geckolinux I get the point. But if I upgrade to Debian 12, then I'm stuck with GNOME 43 which I do not like. Removing it and installing KDE Plasma is nothing short of making a big mess. If I upgrade in that fashion, then I'll need to clean install the KDE spin and upgrade it to Debian 12.

**Reply by geckolinux** [reply]
@banzaigtv Hi there, sorry but I still don't understand. If you don't like something about Debian 12 then you can continue to use your current SpiralLinux installation and still get security updates for years from Debian 11. If you don't like Debian 12 then you won't like SpiralLinux 12 either, since it uses the exact same packages. On the other hand, if you do want Debian 12 with the SpiralLinux customizations then you can quickly and smoothly upgrade it.

**Reply by banzaigtv** [reply]
@geckolinux Well, I'll stay on the Debian 11 base for now since GNOME 3.38 works quite well without crashes and supports VirtualBox. I just want what's best for this computer. As for a 9-year-old laptop I have, which is currently on MX Linux, I may be interested in getting it on SpiralLinux and upgrading that to Debian 12. Will the LXQt spin be the best fit for it? It is a low-spec machine by today's standards. I never used LXQt as a daily driver before, but I will be willing to give it a try since I have tested it on a virtual machine.

**Reply by geckolinux** [reply]
@banzaigtv LXQt should work well, but also the Mate and XFCE versions should be just as light. It depends on your personal preferences.

**Reply by banzaigtv** [reply]
@geckolinux I know that LXQt is featherweight compared to XFCE, but thanks. The KDE spin is within reach, but it might be a bit too much for that old machine to handle.

**Reply by comicallycluttered** [reply]
> Nice! Thanks for saying so.
One potential bottleneck will be the https://fasttrack.debian.net/ repo, which is needed for VirtualBox guest support. (I personally hate VirtualBox, but it's what everyone uses and expects to work out-of-the-box.) I'm not sure how long it will take them to get the Fast Track repos up for Bookworm, but it will eventually be available.

I don't know if they actually work or have all the necessary packages (which might explain why it's not on the main site), but the FastTrack Bookworm repo appears to be available now: https://fasttrack.debian.net/debian-fasttrack/dists/bookworm-fasttrack/

**Reply by geckolinux** [reply]
It's been available since the initial release of Bookworm, but no VirtualBox packages yet.

**Reply by comicallycluttered** [reply]
Ah, okay. Thought that might be the issue. (Didn't have the opportunity to check the package list myself, so apologies for that.)

Thanks for the quick response, though! Hopefully it gets added sometime soon, but at least upgrading from Bullseye is easy enough to do in the meantime.

**Reply by fredjoseph** [reply]
I didn't check but it seems the package should be available now. The fasttrack ticket is closed [Issue-51](https://salsa.debian.org/fasttrack-team/support/-/issues/51)

**Comment by methanoid** [comment]
No but your download the files from website and then you need to use
LiveBuild to do it.

**Comment by zparihar** [comment]
This is pretty awesome. But got a question, is Spiral Linux still planning
releasing ISOs for Bookworm?

**Reply by geckolinux** [reply]
https://github.com/orgs/SpiralLinux/discussions/256#discussioncomment-6148061

**Reply by newbie992** [reply]
I tried to upgrade to debian 12, but nothing works. It is impossible. The synaptic does not accept the new repos. Trying to upgrade from cli (using several tutorials) doesn't work either.
Is there a tutorial for spiral-linux how to upgrade to bookworm for people like me who did not study computer-sciences ?

**Reply by geckolinux** [reply]
@newbie992 
https://github.com/SpiralLinux/SpiralLinux-project/wiki#upgrading-to-a-major-new-debian-stable-release

**Reply by newbie992** [reply]
Thank you. 
bookworm was installed , but I thought it had not worked properly. 
Everything is fine now.

**Comment by zparihar** [comment]
Release it as a BETA... Because if distro testers give it a try in
virtualBox, they may give it a bad review. And I think distro testers are
important to the success of this Spiral Linux project.

**Comment by banzaigtv** [comment]
I would like to see an ISO release for the Bookworm base soon. For VirtualBox, it has been discovered that installing the Ubuntu 22.04 deb package from the Oracle website actually installs VirtualBox on Debian 12. But it has been said that installing Ubuntu packages on Debian is generally not recommended since mixing repositories will create an unstable franken-distro. Also, since GNOME 43 is a total abomination, if I clean install this distro when it's based on Debian 12, then I would install the KDE version. I want to point out that when viewing media file info on Dolphin, the details tab is missing. It seems to be a bug which is not an issue on other distros I have tested, including upstream Debian 12. If you could fix that, that would be great. Also, can you please pre-install Wayland on the KDE version?

**Reply by geckolinux** [reply]
Hi there, the issue isn't with installing VirtualBox for using it as a host, but rather the VirtualBox kernel modules and utilities that make it integrate properly as a _guest_.

> But it has been said that installing Ubuntu packages on Debian is generally not recommended since mixing repositories will create an unstable franken-distro.

Right, and I also don't want to include 3rd party VirtualBox repos in SpiralLinux either.

As a temporary workaround, the `7.0.8-dfsg-2` DEBs from this Debian Unstable snapshot are working for me on a SpiralLinux system upgraded to Debian 12.
http://snapshot.debian.org/archive/debian/20230610T205311Z/pool/contrib/v/virtualbox/

> I want to point out that when viewing media file info on Dolphin, the details tab is missing. It seems to be a bug which is not an issue on other distros I have tested, including upstream Debian 12

Please open a bug report with screenshots from another distro that shows what is missing.

> Also, can you please pre-install Wayland on the KDE version?

Nope, I don't use or support Wayland, it's not ready yet for mainstream use.

**Comment by banzaigtv** [comment]
> > I want to point out that when viewing media file info on Dolphin, the details tab is missing. It seems to be a bug which is not an issue on other distros I have tested, including upstream Debian 12
> 
> Please open a bug report with screenshots from another distro that shows what is missing.
> 
I submitted the bug report.

**Reply by sutram** [reply]
@geckolinux I am quite excited about trying out SpiralLinux with Debian 12 and it seems that Virtualbox 7.0.10 now supports Debian 12: https://www.virtualbox.org/wiki/Linux_Downloads. Would it be possible to bring out new SpiralLinux images based on Debian 12? Thanks!

**Reply by geckolinux** [reply]
Hi there @sutram , SpiralLinux doesn't use VirtualBox from virtualbox.org, but rather what the Debian project itself packages, which are not available yet.
I would encourage you to [upgrade](https://github.com/SpiralLinux/SpiralLinux-project/wiki#upgrading-to-a-major-new-debian-stable-release) your current SpiralLinux installation, as that will get you the exact same thing as a new installation from a future SpiralLinux ISO built from Debian 12.

**Reply by sutram** [reply]
Oh I see. Upgrades are not supported officially until the first point release (in this case 12.1?). I think it is due sometime in July. I can wait until then.

**Reply by geckolinux** [reply]
@sutram No, you can upgrade it right now. SpiralLinux _is_ Debian, and it is meant to be installed once from the ISO and then infinitely updated directly from official Debian online sources, without re-installing from a new ISO.

**Reply by sutram** [reply]
Ok, sounds good - thanks

**Comment by JamesShriner** [comment]
@geckolinux ....you posted on the forum Bruno's All Things Linux awhile ago, and interacted with some of us about Spiral Linux.  I asked about including PenguinsEggs as a remastering software to create custom ISO remixes of Spiral; similar to Remastersys and Systemback.  You indicated you use Debian Live Build, but would be interested in how the PenguinsEggs exercise turns out.  

I followed your advice and upgraded my Spiral 11.2 LXQT VM into Spiral 12.1 LXQT and commenced to customizing.  You may be pleased to know that I now have a custom ISO of Spiral LXQT that is perfect for ME, and ready for deployment on my home network.  I'm prepared to say it was a success, but with caveats.  

Perhaps with your input and smoothing some rough edges the PenguinsEggs developers has experienced with Spiral, I would definitely like to see PenguinsEggs included in Spiral, much like MX-Linux includes a custom remaster tool.  IMO, that is a big reason for MX's rise to the top of the distro charts (along with a polished UI on a supremely stable Debian base!).  Now that Debian allows non-free at installation, the ability to remaster one's custom Spiral install into a backup ISO, or for homogenous deployment across multiple machines COULD be a distinguishing feature for Spiral, IMO.  I don't want to hijack this thread, but it seemed like a good place to mention it.  Check out the BATL discussion to bring yourself up to speed and if you think it's worthy of discussion here, I'm happy to start a new thread for remastering Spiral into custom ISOs.  

I'm not a developer....just a hacker who keeps poking around until success is found.  But I'm happy to share my learnings for the greater good.  If you don't want to go down that road, it's fine.....it is YOUR project, and I got what I wanted.  But if I can help "pay it forward" in the spirit of Linux, and help to improve Spiral in the process, I'm happy to share what I learned.  Your thoughts?

**Comment by methanoid** [comment]
@JamesShriner  Devuan's Refracta Snapshot I guess is similar... do you have a link to this Penguins Eggs, please? My Google-fu failed

**Reply by JamesShriner** [reply]
I've used Refracta also.  Even though successful, Refracta was a little "quirky" for me and I had problems with installation....installing bootloader if I recall correctly.  I can't say it was a Refracta issue....might've been an "I did something wrong" issue.....but the Refracta developer wasn't able to help.  I think I ended up chrooting into the failed install and installing grub from there.  It's been awhile....but it left a bad taste in my mouth.

Remastersys was the GOLD standard, IMO.  And when it was abandoned, several folks forked it....but they were all "buggy" for me.  I'd say that I am the common problem across all these "buggy" tools, so maybe it's ME; except I NEVER had a problem with Remastersys.  I had that tool "mastered", LOL!  Been looking for something similar ever since.  PenguinsEggs is the closest I have found, except it has no GUI.  Even though it's CLI, it's very user-friendly.  And it's early in the development, so I can only imagine it getting better & better.  Here's a link:

https://penguins--eggs-net.translate.goog/docs/Tutorial/eggs-users-guide?_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=wapp&_x_tr_hist=true

You can install via PPA, but I don't like PPAs on Debian.  I just installed the DEB package, read the manual, and started tinkering until it became familiar.  The developer is a Debian user, and it started out as a Debian tool, but he's now able to support Arch-based distros, like Manjaro, Garuda, Endeavour, etc...  with a package in the AUR.  No RPM distros yet, though.

My caveat with Penguins Eggs and Spiral is that if you create a custom Spiral ISO on a UEFI/GPT formatted partition, it only installs on a UEFI/GPT partition; as Calamares installation fails on the bootloader.sh script trying to remove grub-efi-amd64 and replace with grub-pc.  I found a workaround, with the developer's help, but this isn't the place for that discussion.  My advice, especially for your first attempt, is to create your custom Spiral ISO on the partition scheme you  expect to install it on.  If you create a custom Spiral on UEFI/GPT, only install it on UEFI/GPT machines.  If you plan on installing to a BIOS/MBR scheme, create your Spiral ISO on that configuration.

Like I said....there's a workaround, and it isn't hard or convoluted....it's just a PITA.  And I'd like you to have a GOOD experience on your first attempt.  FWIW....

Good luck!

**Reply by methanoid** [reply]
OMG that is just so dumb (IMHO). Google tran makes an already wacky (stupid really) naming system (mom,dad,eggs,krill ?!??!?!) even harder to decipher.  The only sensible documentation is found in English here   https://github.com/pieroproietti/penguins-eggs#readme

But I can't be arsed to expend mental energy translating "krill" into something sensible in my head. I'll stick to "live-build" or follow YouTube videos for that (and Refacta). Good luck searching for videos to help with this one. I'm not trawling pages of penguin videos to see if any guides exist!!!

**Reply by JamesShriner** [reply]
yes, the developer appears to be italian and the english is a little "stilted" at times, but everything you need is in that README file.  Once installed, all you really need is "sudo eggs dad" to provide some "variables" for your ISO template (i.e. name of ISO, name of live-user, passwords, etc...); "sudo eggs produce" to actually create the ISO; and then "sudo eggs kill" to cleanup the ISO files from your system after you are satisfied with your ISO.

It's not that hard.  But if you prefer Refracta or Live-build, that's cool.....it's Linux, and we have options.  One size does NOT fit all.  Nothing shoved down our throats like other OS.  Just another tool for the toolbox.  Some may find it useful, like I did.
